### PR TITLE
Implement task manager API with a documentation

### PR DESCRIPTION
**Pull request by Mvamd12**
This pull request delivers a fully functional Task Manager API using FastAPI. It includes endpoints for creating, retrieving, updating, and deleting tasks, along with improved error handling to manage invalid requests. Additionally, it introduces a comprehensive README.md that outlines the setup process, API endpoint details, and usage instructions. This update provides a robust solution for task management and clear documentation to facilitate easy integration and contribution.

### CODE REVIEW

**Comment by Momen-G-Ar** [comment]
As the README.md file requires: why didn't you use more than one module when creating your API?
Please separate and organize your code into different modules

File: MohammadBallal/main.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+class Task(BaseModel):
+    task_id: UUID= uuid4()  # Automatically generates a unique ID for each task
+    title: str                                
+    description: str = "No description provided"                           
+    due_date: date                            
+    status: Status 
+    priority: Priority = None
+
+    
+tasks= []
```
Comment (Momen-G-Ar) [inline MohammadBallal/main.py:32]: In my opinion, using `dict` is better and easier that using two lists

File: MohammadBallal/main.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+from pydantic import BaseModel
+from uuid import UUID, uuid4
+from datetime import date
+from enum import Enum
+
+
+
+app = FastAPI()
+
+# Using enum for status and priority
```
Comment (Momen-G-Ar) [inline MohammadBallal/main.py:11]: As i mentioned before, please write classes and models in another file

**Review by Momen-G-Ar — CHANGES_REQUESTED** [review]
Please do the required

File: MohammadBallal/main.py (hunk 2 above)
Comment (Mvamd12) [reply MohammadBallal/main.py]: I handled the matter based on the tutorial you presented, and you mentioned that in the coming days, we will deal with code organization and modules, BTW The required change is done.

File: MohammadBallal/main.py (hunk 1 above)
Comment (Mvamd12) [reply MohammadBallal/main.py]: I understand you, and if this is necessary I will make the adjustment if you want.

File: MohammadBallal/app/main.py
Comment (Momen-G-Ar) [inline MohammadBallal/app/main.py:1]: I don't think it's necessary here, but learning it and using it in the upcoming tasks is a great idea. Great 🔥

Comment (Mvamd12) [reply MohammadBallal/app/main.py:1]: Thank you, I appreciate your advice.

File: MohammadBallal/app/routes.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    # Search for the task in both lists
+    for task_list in (urgent_tasks, tasks):
+        for task in task_list:
+            if task.task_id == task_id:
+                return {"task": task}
+
+    raise HTTPException(status_code=404, detail="Task not found")
+
+# Update a Task by ID
+@router.put("/tasks/{task_id}/")
```
[the remark this answers is not in the document]
Comment (Mvamd12) [reply MohammadBallal/app/routes.py:46]: The changes is done. Please review the code.

**Review by FaresAbuIram — COMMENTED** [review]
Please add the Class for Task and UrgentTask. Add the constructor for both and use them in your arrays, as this is the main class. The base model class is not sufficient for processing; we only use it to pass correct data to our endpoints.

File: MohammadBallal/app/routes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Momen-G-Ar) [reply MohammadBallal/app/routes.py:46]: Great work, but it is preferred to make them into one variable with one Type:
for example, make a class type for optional fields and assign it to a single variable

And after that I want another update depending on it, I will tell you about it after pushing the new change

[the remark this answers is not in the document]
Comment (Mvamd12) [reply MohammadBallal/app/routes.py:47]: The changes have been made, and I expect that the additional issue which I wasn't sure about has been resolved.

**Comment by Mvamd12** [comment]
> Please add the Class for Task and UrgentTask. Add the constructor for both and use them in your arrays, as this is the main class. The base model class is not sufficient for processing; we only use it to pass correct data to our endpoints.

As I mentioned in the stand-up meeting, I initially handled the issue as a limited task for a specific purpose. However, I have fixed it now. Thank you.

File: MohammadBallal/app/routes.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        raise HTTPException(status_code=404, detail="Task not found")
+
+    # Remove the task from its current list
+    if task_to_update in tasks:
+        tasks.remove(task_to_update)
+    elif task_to_update in urgent_tasks:
+        urgent_tasks.remove(task_to_update)
+
+    # Update task fields
+    if update_data.title is not None:
```
Comment (Momen-G-Ar) [inline MohammadBallal/app/routes.py:65]: What happens if the attributes are more? 10? 20?
are u sure to use this way?
let us discuss the problem: you do the same operations for all properties for an object, I think it is better to use a loop to iterate over all of them and handle these operations.

File: MohammadBallal/app/models.py
Comment (Momen-G-Ar) [inline MohammadBallal/app/models.py:1]: What is the difference between Task and TaskModel? ? ?

File: MohammadBallal/app/routes.py (hunk 4 above)
Comment (Mvamd12) [reply MohammadBallal/app/routes.py]: You’re right. I treated the attributes as if they were limited, but as you mentioned it’s better to use a loop when there are many attributes. It's done.

File: MohammadBallal/app/models.py
Comment (Mvamd12) [reply MohammadBallal/app/models.py:1]: In simple manner, Task is for represinting task data and TaskModel for validation and pass data to endpoint

File: MohammadBallal/app/routes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Momen-G-Ar) [reply MohammadBallal/app/routes.py:47]: Great work

File: MohammadBallal/app/models.py
Comment (Momen-G-Ar) [inline MohammadBallal/app/models.py:1]: In the next time, separate classes and models each in an independent file _`(better)`_

**Review by Momen-G-Ar — APPROVED** [review]

File: MohammadBallal/app/models.py
Comment (Mvamd12) [reply MohammadBallal/app/models.py:1]: Ok thank you, I will work on it.

**Pull request merged**